src/raft/raft.go

```diff
@@ -326,33 +311,51 @@ func (rf *Raft) HeartBeat(args *HeartBeatRequest, reply *HeartBeatReply) {
 			for i, v := range rf.log[args.PrevLogIndex+1 : rf.lastLogIndex()+1] {
 				if i < len(args.Entries) && v.Term != args.Entries[i].Term {
 					rf.log = rf.log[:args.PrevLogIndex+1+i] // here use Index Not Term!...
+					rf.persist()                            // at HeartBeat log updates
```

Comment:
These places for speed-up follower log correction are important for passing `TestFigure8Unreliable2C`